pkg/gcs/gcsClient.go

```diff
@@ -0,0 +1,216 @@
+package gcs
+
+import (
+	"context"
+	"io"
+	"os"
+	"path/filepath"
+
+	"cloud.google.com/go/storage"
+	"github.com/SAP/jenkins-library/pkg/log"
+	"github.com/pkg/errors"
+	"google.golang.org/api/iterator"
+	"google.golang.org/api/option"
+)
+
+// Client is an interface to mock gcsClient
+type Client interface {
+	UploadFile(bucketID string, sourcePath string, targetPath string) error
+	DownloadFile(bucketID string, sourcePath string, targetPath string) error
+	ListFiles(bucketID string) ([]string, error)
+	Close() error
+}
+
+// gcsClient provides functions to interact with google cloud storage API
+type gcsClient struct {
+	context       context.Context
+	envVars       []EnvVar
+	client        storage.Client
+	clientOptions []option.ClientOption
+	openFile      func(name string) (io.ReadCloser, error)
+	createFile    func(name string) (io.WriteCloser, error)
+}
+
+// EnvVar defines an  environment variable incl. information about a potential modification to the variable
+type EnvVar struct {
+	Name     string
+	Value    string
+	Modified bool
+}
+
+type gcsOption func(*gcsClient)
+
+func WithEnvVars(envVars []EnvVar) gcsOption {
```

Comment:
please provide comment for exported functions

Reply:
sure